internal/pool/pool.go

```diff
@@ -179,6 +179,7 @@ func (p *ConnPool) dialConn(ctx context.Context, pooled bool) (*Conn, error) {
 		return nil, err
 	}
 
+	internal.Instruments.NewConnectionsCount.Add(ctx, 1)
```

Comment:
Did we want to include any labels with this?

Reply:
Yes I think that would be a good idea. 

Before we added any labels though, we wanted to show @vmihailenco the high-level changes to the code that would be involved when adding metrics